components/datetime/src/fields/length.rs

```diff
@@ -3,12 +3,16 @@
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 use std::convert::TryFrom;
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum LengthError {
     TooLong,
 }
 
 #[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "provider_serde",
```

Comment:
> I think the "serde" feature should enable serde for library types, and "provider_serde" should enable it for data structs.

Ah, that kind of separation makes sense.

I filed #556 to document the features, so that we can have some shared understanding of the intent of various features. I've found myself struggling with how to think about them, especially in the meta crate context.

> On the other hand, this enum does not get exported, so maybe it's okay to put it on "provider_serde". Thoughts?

I like being able to write code with how it's being used, so I'm inclined to keep it provider_serde. Although this choice should probably be documented.